### PR TITLE
OLS-81: switchover to CI-built RAG; OLS-355: Faiss

### PR DESCRIPTION
**Pull request by syedriko**
## Description

- Switched build of the RAG'ed OLS image to RAG content built by CI into quay.io/openshift/lightspeed-rag-content
- OLS now expects the RAG vector database to be in Faiss format
- Removed RAG build scripts as they've migrated to https://github.com/openshift/lightspeed-rag-content

## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes # https://issues.redhat.com/browse/OLS-81, https://issues.redhat.com/browse/OLS-355

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

File: Containerfile.rag [hunk 1]
```diff
@@ -1,21 +1,18 @@
 # vim: set filetype=dockerfile
-
-# port, commands, and other details inherited from the base image
 ARG OLS_VERSION=latest
+ARG OCP_VERSION=4.15
+ARG LIGHTSPEED_RAG_CONTENT_DIGEST=sha256:eb3472dd675cc79b02e4d15b53b12c9c365d6c2335a6026f55de443dd95c902f
+
+FROM quay.io/openshift/lightspeed-rag-content@${LIGHTSPEED_RAG_CONTENT_DIGEST} as lightspeed-rag-content
```
Comment (vishnuchalla) [inline Containerfile.rag:6]: Just wondering, if this image already has embeddings stored in vector db or just the plain docs? I hope it only has embeddings and that too are generated as a Day 1 operation (.i.e just like a one time activity)

Comment (syedriko) [reply Containerfile.rag:6]: The lightspeed-rag-content image has the embedding model and the vector db it. The plaintext docs are in the openshift/lightspeed-rag-content repo.

Comment (bparees) [reply Containerfile.rag:6]: that image has the generated embeddings.

the idea is that we will periodically generate a new embedding image and then update the SHA in this file to pick up the new embeddings.

File: Containerfile.rag [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
+ARG OLS_VERSION
```
Comment (bparees) [inline Containerfile.rag:7]: i don't understand the purpose of this line?

File: Containerfile.rag [hunk 3]
```diff
@@ -1,21 +1,18 @@
...
-    && cd vector-db \
-    && wget -q $RAG_INDEX \
-    && unzip -qq local.zip \
-    && mv local ocp-product-docs \
-    && rm -f local.zip \
-    && microdnf remove -y unzip wget
-
-COPY scripts/download_embeddings_model.py .
-RUN python3.11 download_embeddings_model.py -l embeddings_model -r BAAI/bge-base-en && rm download_embeddings_model.py
+COPY --from=lightspeed-rag-content /rag/vector_db/ocp_product_docs/${OCP_VERSION} ./vector_db/ocp_product_docs/${OCP_VERSION}
```
Comment (bparees) [inline Containerfile.rag:17]: how are we going to support multiple ocp versions like this?  ultimately the emebddings image should/will contain embeddings for multiple OCP docs versions, and we should just copy them all into our OLS image.

**Review by bparees — COMMENTED** [review]
couple small items from me but the fact that this is failing e2e tests means something deeper is broken that needs to be looked at

**Comment by syedriko** [comment]
> couple small items from me but the fact that this is failing e2e tests means something deeper is broken that needs to be looked at

Already did :) This
https://github.com/openshift/lightspeed-service/blob/578a030ed1eecfbb193077dfb8db91e695265d38/tests/scripts/test-e2e-standalone.sh#L36
downloads the old RAG, in SimpleVectorStore format.
But, it doesn't download the embedding model. There needs to be "embeddings_model_path" in https://github.com/openshift/lightspeed-service/blob/578a030ed1eecfbb193077dfb8db91e695265d38/tests/config/singleprovider.e2e.template.config.yaml#L11

File: Containerfile.rag (hunk 3 above)
Comment (syedriko) [reply Containerfile.rag:17]: Are you saying to just drop any mention of of OCP_VERSION from Container.rag and
```
COPY --from=lightspeed-rag-content /rag/vector_db/ocp_product_docs ./vector_db/ocp_product_docs
```
?

File: Containerfile.rag (hunk 2 above)
Comment (syedriko) [reply Containerfile.rag:7]: Generatlly that's passing an ARG from one stage to another, but, true, not needed in this case.

File: Containerfile.rag (hunk 3 above)
Comment (bparees) [reply Containerfile.rag:17]: yeah

Comment (syedriko) [reply Containerfile.rag]: Done

File: Containerfile.rag (hunk 2 above)
Comment (syedriko) [reply Containerfile.rag]: Done

File: Containerfile.rag (hunk 1 above)
Comment (vishnuchalla) [reply Containerfile.rag:5]: Understood. Thanks for confirming.

File: tests/scripts/test-e2e-standalone.sh [hunk 4]
```diff
@@ -33,20 +33,20 @@ TMPDIR=$(mktemp -d)
...
 # configure feedback storage location
 export FEEDBACK_STORAGE_LOCATION="$ARTIFACT_DIR/insights/feedback"
 echo "Feedback storage location: $FEEDBACK_STORAGE_LOCATION"
 
+# temp directory for index store
+RAG_TMP_DIR=$(mktemp -d)
+# Download index store
+export RAG_INDEX_DIR="$RAG_TMP_DIR/vector_db/ocp_product_docs/4.15"
+export RAG_EMBEDDINGS_MODEL_DIR="$RAG_TMP_DIR/embeddings_model"
+podman create --replace --name tmp-rag-container quay.io/openshift/lightspeed-rag-content@sha256:eb3472dd675cc79b02e4d15b53b12c9c365d6c2335a6026f55de443dd95c902f true
```
Comment (bparees) [inline tests/scripts/test-e2e-standalone.sh:45]: https://access.redhat.com/solutions/6964609

we might be stuck :(

Comment (syedriko) [reply tests/scripts/test-e2e-standalone.sh:45]: The solution with an OLS+RAG+test, the way this branch was in the morning, should I think work.

Comment (bparees) [reply tests/scripts/test-e2e-standalone.sh:45]: yeah sorry i just meant stuck in terms of direct podman usage.

**Pull request merged**